src/biome/text/dataset_readers/mixins.py

```diff
@@ -97,17 +100,17 @@ def build_textfield(
             Either a TextField or a ListField containing the record.
             Returns None if `data` is not a str or a dict.
         """
-        if not isinstance(data, (str, list, dict)):
+        if not isinstance(data, (str, Iterable, dict)):
```

Comment:
```suggestion
        if not isinstance(data, (str, Iterable)):
```
should be sufficient, since dict is also an iterable